f2/src/forms/ContactInfoForm.js

```diff
@@ -1,162 +1,160 @@
-        </div>
-      ) : null}
-      <Form
+      <Formik
         initialValues={contactInfo}
-        onSubmit={localOnSubmit}
-        validate={(values) => {
-          if (fyiForm) {
-            return fyiValidate(values)
+        onSubmit={async (values, { setErrors }) => {
+          const errors = !fyiForm
+            ? onSubmitValidation(values)
+            : onSubmitFYIValidation(values)
+          if (errors.fields) {
+            setErrors(errors.fields)
           } else {
-            return validate(values)
+            props.onSubmit(values)
           }
         }}
-        render={({
-          handleSubmit,
-          values,
-          errors,
-          submitFailed,
-          hasValidationErrors,
-        }) => (
-          <Stack
-            as="form"
-            onSubmit={handleSubmit}
-            shouldWrapChildren
-            spacing={6}
-          >
-            {submitFailed && hasValidationErrors ? (
-              <ErrorSummary>
-                <Trans id="contactinfoPage.hasValidationErrors" />
-              </ErrorSummary>
-            ) : null}
-            {fyiForm ? (
-              <Flex direction="row" align="center" wrap="wrap" mb={10}>
-                <P w="100%">
-                  <Trans id="contactinfoPage.skipInfo" />
-                </P>
-                <Button
-                  as={ReactRouterLink}
-                  fontSize={{ base: 'lg', md: 'xl' }}
-                  color="black"
-                  variant="solid"
-                  variantColor="gray"
-                  bg="gray.400"
-                  borderColor="gray.500"
-                  to="/confirmation"
-                  textAlign="center"
-                >
-                  <Trans id="locationinfoPage.skipButton" />
-                  <Icon
-                    focusable="false"
-                    ml={2}
-                    mr={-2}
-                    name="chevron-right"
-                    size="28px"
+      >
+        {({ handleSubmit, handleChange, handleBlur, errors, submitCount }) => (
+          <Form onSubmit={handleSubmit}>
+            <Container>
+              <Row className="form-section">
+                {Object.keys(errors).length > 0 && (
+                  <ErrorSummary
+                    errors={createErrorSummary(errors)}
+                    submissions={submitCount}
+                    title={<Trans id="contactinfoPage.hasValidationErrors" />}
                   />
-                </Button>
-              </Flex>
-            ) : null}
-            <Field
-              name="fullName"
-              label={<Trans id="contactinfoPage.fullName" />}
-              errorMessage={<Trans id="contactinfoForm.fullName.warning" />}
-              component={Input}
-              required={!fyiForm}
-            />
+                )}
+
+                {errors && errors.enterContactDetails && (
+                  <P color="#dc3545" fontSize="1.25rem" marginBottom="0.5rem">
```

Comment:
Errors should be displayed above the related field. This will display all errors at the top of the form below the error summary.

Reply:
The errors are on the top now,